### DISCUSSION TITLE
[8 Giugno 2023] Modifica CPU

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/architettura-degli-elaboratori#73] [Modifiche architettura MIPS (M-Z)]
![image](https://github.com/sapienzastudentsnetwork/archelab2223/assets/66442720/9a70d8e9-aef8-410a-bd89-d5ba8ec7d1e4)
![image](https://github.com/sapienzastudentsnetwork/archelab2223/assets/66442720/6c56901c-c908-48fa-a666-986a3b8e92b2)
![image](https://github.com/sapienzastudentsnetwork/archelab2223/assets/66442720/3e5ca79e-3815-413e-800d-b7881be9a25b)

**Comment by AglaiaNorza** [comment]
dovrebbe avere senso
![image](https://github.com/sapienzastudentsnetwork/architettura-degli-elaboratori/assets/75022526/75e23ca7-a9bb-4c8d-bd77-5fbc9dd135e9)

per la condizione del write register ho settato regdst a 0 e l'ho messo in OR con una AND tra il bit che controlla la condizione dell'esercizio e il segnale lwsf (così da non settare il "regdst" a 1 per sbaglio in altre istruzioni)

per i submem ho usato la ALU per non creare altri sommatori visto che sarebbe rimasta inutilizzata

![image](https://github.com/sapienzastudentsnetwork/architettura-degli-elaboratori/assets/75022526/735f076d-8d5b-4b1a-aaae-97a093d7c7c5)

**Reply by matypist** [reply]
> dovrebbe

Confermo la proposta di soluzione di @AglaiaNorza, tutto chiaro e corretto 👍

> per la condizione del write register ho settato regdst a 0 e l'ho messo in OR con una AND tra il bit che controlla la condizione dell'esercizio e il segnale lwsf (così da non settare il "regdst" a 1 per sbaglio in altre istruzioni)

Molto interessante soprattutto l'approccio risolutivo al punto 1 (l'architettura per come l'ha modificata), non avevo pensato alla possibilità di risfruttare il MUX con una porta OR e RegDst impostato a 0

**Comment by rimaout** [comment]
<img width="790" alt="image" src="https://github.com/sapienzastudentsnetwork/architettura-degli-elaboratori/assets/116072651/029a7e02-c763-434a-8229-3dd906b6eea4">
